token.go

```diff
@@ -9,8 +10,22 @@ import (
 // the key for verification.  The function receives the parsed, but unverified
 // Token.  This allows you to use properties in the Header of the token (such as
 // `kid`) to identify which key to use.
+//
+// The returned interface{} may be a single key or a PublicKeyset containing
+// multiple keys.
 type Keyfunc func(*Token) (interface{}, error)
 
+// PublicKey represents a generic public key interface.
+type PublicKey interface {
+	crypto.PublicKey | []uint8
```

Comment:
Should we also add `unsafeNoneMagicConstant` (for the `none` algorithm) to this? Then we would include all the allowed key types. Might make sense, if we ever change the `Keyfunc` function signature at some point.

Reply:
I've been going back and forth on this one. Since `VerificationKey` is used in the key set and thus in verification, what does it mean to have a "rotating" key of none.

I wonder if this is something we can punt on and add when we do decide to expand the `VerificationKey` in more places? Afaics adding this later isn't a breaking change.

Reply:
Going to sleep on this one, and see if there's any feedback. But maybe you're right, adding it is the correct thing to do here so `VerificationKey` can represent all the keys (in a type-safe way) that we support.

Reply:
> Going to sleep on this one, and see if there's any feedback. But maybe you're right, adding it is the correct thing to do here so `VerificationKey` can represent all the keys (in a type-safe way) that we support.

Yes, that was my original thinking to have all of them in one place.

However, in re-reading the RFCs again, I am beginning to come to the conclusion that we might NOT even "need" to support the `none` algorithm. So this might also be the point in deciding to drop this in the future; or at least deprecate this in the future and NOT include this in this interface here.

Reply:
I think this PR accomplishes what we set out,

1. Add key rotation support
2. In a type-safe way
3. While adding a well-defined type for future improvements

My current thought is .. let's punt on adding (if ever) the `unsafeNoneMagicConstant` type. I'd like to understand what the practical cases are for it, and/or whether there is a true community need for this.

Starting with a narrow scope won't tie our hands in the future. But afaics we can expand the interface in the future if necessary.